### PR TITLE
Add apidoc_separate_modules option

### PR DESCRIPTION
**Pull request by zaneb**
Give users the option to have apidoc generate a page-per-module instead
of a page-per-package.

### CODE REVIEW

**Comment by stmcginnis** [comment]
This would help keep Cinder docs consistent with the existing doc output. I have not had a chance to test it out yet, but the code looks good, and just want to state support for the idea.

File: sphinxcontrib/apidoc/__init__.py [hunk 1]
```diff
@@ -28,5 +28,6 @@ def setup(app):
     app.add_config_value('apidoc_module_dir', None, 'env', [str])
     app.add_config_value('apidoc_output_dir', 'api', 'env', [str])
     app.add_config_value('apidoc_excluded_paths', [], 'env', [[str]])
+    app.add_config_value('apidoc_separate_modules', False, 'env', [[bool]])
```
Comment (stephenfin) [inline sphinxcontrib/apidoc/__init__.py]: s/[[bool]]/[bool]/

(this is a type check, and what's there means "list of bools" when I guess we just want a single bool)

Comment (stephenfin) [reply sphinxcontrib/apidoc/__init__.py]: Went ahead and fixed this. Also added some tests along the way.

**Comment by zaneb** [comment]
Awesome, thanks @stephenfin

**Pull request merged**